### PR TITLE
updating release update to be macOS sed compatible

### PR DESCRIPTION
**Pull request by beta-ziliani**
https://github.com/crystal-lang/crystal/pull/12260 was generated with the changes in this PR, on macOS

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

File: scripts/release-update.sh [hunk 1]
```diff
@@ -12,29 +12,29 @@ set -eu
 CRYSTAL_VERSION=$1
 
 # Edit PREVIOUS_CRYSTAL_BASE_URL in .circleci/config.yml
-sed -i -E "s|[0-9.]+/crystal-[0-9.]+-[0-9]|$CRYSTAL_VERSION/crystal-$CRYSTAL_VERSION-1|g" .circleci/config.yml
+sed -i '' -E "s|[0-9.]+/crystal-[0-9.]+-[0-9]|$CRYSTAL_VERSION/crystal-$CRYSTAL_VERSION-1|g" .circleci/config.yml
```
Comment (Fryguy) [inline scripts/release-update.sh:15]: Is this script only run on MacOS?  `-i ''` doesn't work on Linux

Comment (Fryguy) [reply scripts/release-update.sh:15]: FWIW, this particular thing is the bane of my existence in my other projects.  In those other projects I do `brew install coreutils` to get the gnu versions of common tools, then temporarily make `sed` point to `gsed` before running any scripts.  Trying to make them work on both OSes can be a nightmare

Comment (beta-ziliani) [reply scripts/release-update.sh:15]: yikes, thanks @Fryguy

Comment (Fryguy) [reply scripts/release-update.sh:15]: If it helps, you can determine if MacOS using the following test in bash: `if [[ "$OSTYPE" == "darwin"* ]]; then ...`

Comment (beta-ziliani) [reply scripts/release-update.sh]: I thought about it, but I decided to test `sed` directly in case it's locally linked to gnu's

**Comment by straight-shoota** [comment]
Sigh, why can't things be simple... 
@beta-ziliani WDYT, would it be reasonable to require the gnu version of sed when running this on Macos?
That would avoid adding more complexity. We're using sed in many other scripts as well.
We could add a check to make sure we have the GNU version and abort otherwise?

**Comment by Fryguy** [comment]
You can also "override" `sed` within the script and do the suggestion that @straight-shoota suggests with something like:

```bash
sed() {
  command sed --version &>/dev/null
  if [ $? -eq 0 ]; then
    command sed "$@"
  else
    gsed "$@"
    # Or get more fancy here and check for gsed and suggest installing it if it doesn't exist.
  fi
}
```

**Comment by beta-ziliani** [comment]
I like @Fryguy suggestion. For the meantime, I'll close this one.

**Pull request closed without merging**